cmake/modules/RootNewMacros.cmake

```diff
@@ -1449,4 +1449,27 @@ function(find_python_module module)
    set(PY_${module_upper}_FOUND ${PY_${module_upper}_FOUND} PARENT_SCOPE)
 endfunction()
 
+#----------------------------------------------------------------------------
+# Generate headers and link them through argparse method
+#----------------------------------------------------------------------------
+function(generateHeaders pythonInput output target)
+     add_custom_command(OUTPUT ${output}
+          BYPRODUCTS ${output} DEPENDS ${pythonInput} ${CMAKE_SOURCE_DIR}/build/misc/argparse2help.py
+          COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/build/misc/argparse2help.py 
+                                       ${pythonInput}
+                                       ${output}
+     )
+     target_sources(${target} PRIVATE ${output})
```

Comment:
We should probably wrap this here like:
```cmake
if (TARGET ${target})
  target_sources(${target} PRIVATE ${output})
endif()
```

This would reduce the calls of the function from the outside.

Reply:
No, I think there is a danger to silently fail to add the file to the target if the target does not exist. If there is no target with the given name, we should fail to configure.